langchain/src/prompts/chat.ts

```diff
@@ -62,7 +62,14 @@ export class MessagesPlaceholder extends BaseMessagePromptTemplate {
   }
 
   formatMessages(values: InputValues): Promise<BaseChatMessage[]> {
-    return Promise.resolve(values[this.variableName] as BaseChatMessage[]);
+    return Promise.resolve([
+      values[this.variableName].length > 0 &&
+      this.variableName.includes("history")
+        ? `Relevant pieces of previous conversation:: ${
```

Comment:
Not sure about hardcoding the prompt at this location. We need to also make sure `BaseChatMessage` is returned instead of a `string`. 

If I understand correctly, this is desired for LLMs which do not support the ChatML syntax? Shouldn't that be the responsibility of the LLM?

Reply:
Well, the bug I encountered was that, while I was using the conversation Agent with memory, the chat_history will be converted to a pure string message, which doesn't not have the type nor text attributes.

It turns out it was due to the placeholder format result.

Due to the limit understanding of the whole project, I tested with this hard coded prompt to verify the bug. Ideally we should create a message type for chat history.